src/test/java/seedu/address/logic/commands/UndoCommandTest.java

```diff
@@ -0,0 +1,178 @@
+        expectedModel.deleteResume(meResume);
+        expectedModel.setResumeToDisplay();
+        expectedModel.commitResumeBook();
+        expectedModel.undoResumeBook();
+
+        CommandResult commandResult =
+                new UndoCommandResult("Undo successfully!", expectedModel.getDisplayType());
+        assertCommandSuccess(new UndoCommand(), model, commandResult, expectedModel);
+    }
+
+    @Test
+    public void execute_addInternship_success() {
+        Internship paypal = new InternshipBuilder(TypicalInternship.PAYPAL).build();
+        model.addInternship(paypal);
+        model.setInternshipToDisplay();
+        model.commitResumeBook();
+        expectedModel.addInternship(paypal);
+        expectedModel.setInternshipToDisplay();
+        expectedModel.commitResumeBook();
+        expectedModel.undoResumeBook();
+
+        CommandResult commandResult =
+                new UndoCommandResult("Undo successfully!", expectedModel.getDisplayType());
+        assertCommandSuccess(new UndoCommand(), model, commandResult, expectedModel);
+    }
+
+    @Test
+    public void execute_sortProject_success() {
+        model.sortProjects(Comparator.comparing(Project::getTime));
+        model.setProjectToDisplay();
+        model.commitResumeBook();
+        expectedModel.sortProjects(Comparator.comparing(Project::getTime));
+        expectedModel.setProjectToDisplay();
+        expectedModel.commitResumeBook();
+        expectedModel.undoResumeBook();
+
+        CommandResult commandResult =
+                new UndoCommandResult("Undo successfully!", expectedModel.getDisplayType());
+        assertCommandSuccess(new UndoCommand(), model, commandResult, expectedModel);
+    }
+
+    @Test
+    public void execute_multipleChanges_success() {
+        // Changes to the model copied from the tests above
+        Resume meResume = new ResumeBuilder(TypicalResume.ME_RESUME).build();
+        model.deleteResume(meResume);
+        model.setResumeToDisplay();
+        model.commitResumeBook();
+        expectedModel.deleteResume(meResume);
+        expectedModel.setResumeToDisplay();
+        expectedModel.commitResumeBook();
+
+        Internship paypal = new InternshipBuilder(TypicalInternship.PAYPAL).build();
+        model.addInternship(paypal);
+        model.setInternshipToDisplay();
+        model.commitResumeBook();
+        expectedModel.addInternship(paypal);
+        expectedModel.setInternshipToDisplay();
+        expectedModel.commitResumeBook();
+
+        model.sortProjects(Comparator.comparing(Project::getTime));
+        model.setProjectToDisplay();
+        model.commitResumeBook();
+        expectedModel.sortProjects(Comparator.comparing(Project::getTime));
+        expectedModel.setProjectToDisplay();
+        expectedModel.commitResumeBook();
+
+        expectedModel.undoResumeBook();
+        CommandResult commandResult1 =
+                new UndoCommandResult("Undo successfully!", expectedModel.getDisplayType());
```

Comment:
When you are sure about the expectedModel's display type, I think it would be better to ItemUtil.blahblah instead of expectedModel.getDisplayType, that way we're sure after the undo, the list shown is correct too.

Reply:
@helloImHai 's suggestion is reasonable; agreed.

Reply:
I specifically left it that way because `displayType` is part of the `VersionedResumeBook` state. 

I have added a new `assertEquals` to ensure `expectedModel.getDisplayType()` returns the correct type in the first place.